src/controllers/dynakube/dtclient_reconciler.go

```diff
@@ -112,14 +112,15 @@ func (r *DynatraceClientReconciler) Reconcile(ctx context.Context, instance *dyn
 	var tokens []tokenConfig
 	if r.PaasToken == "" {
 		tokens = []tokenConfig{{
-			Type:  dynatracev1beta1.APITokenConditionType,
-			Key:   dtclient.DynatraceApiToken,
-			Value: r.ApiToken,
-			Scopes: []string{dtclient.TokenScopeDataExport,
-				dtclient.TokenScopeInstallerDownload,
-			},
+			Type:      dynatracev1beta1.APITokenConditionType,
+			Key:       dtclient.DynatraceApiToken,
+			Value:     r.ApiToken,
+			Scopes:    []string{dtclient.TokenScopeInstallerDownload},
 			Timestamp: &r.status.LastAPITokenProbeTimestamp,
 		}}
+		if !instance.FeatureDisableHostsRequests() {
+			tokens[0].Scopes = append(tokens[0].Scopes, dtclient.TokenScopeDataExport)
+		}
```

Comment:
This change would not work if both tokens are provided, since `r.PaasToken` would not be empty.

I'd suggest moving these lines to what is currently line 142, i.e. before `if instance.KubernetesMonitoringMode() &&
		instance.FeatureAutomaticKubernetesApiMonitoring() {` and removing the scope from the first token in the else-branch, i.e. line 131,  to solve this issue.
Please also add an unit test to check for this case.

Otherwise, lgtm

Reply:
done :heavy_check_mark: